src/Reflection/ReflectionClass.php

```diff
@@ -241,70 +283,33 @@ function (ReflectionMethod $method) : array {
                 },
                 $this->getTraits(),
             ),
-            ...array_map(
-                static function (ReflectionClass $ancestor) : array {
-                    return $ancestor->getMethods();
-                },
-                array_values(array_merge(
-                    array_filter([$this->getParentClass()]),
-                    $this->getInterfaces(),
-                )),
-            ),
         );
     }
 
     /**
-     * @return ReflectionMethod[]
+     * @return list<ReflectionMethod>
      */
-    private function createMethodsFromTrait(ReflectionMethod $method) : array
+    private function getMethodsFromInterfaces() : array
     {
-        $traitAliases     = $this->getTraitAliases();
-        $traitPrecedences = $this->getTraitPrecedences();
-
-        $methodAst = $method->getAst();
-        assert($methodAst instanceof ClassMethod);
-
-        $methodHash = $this->methodHash($method->getDeclaringClass()->getName(), $method->getName());
-
-        $aliases = [];
-        foreach ($traitAliases as $aliasMethodName => $traitAliasDefinition) {
-            if ($methodHash !== $traitAliasDefinition) {
-                continue;
-            }
-
-            $aliases[] = ReflectionMethod::createFromNode(
-                $this->reflector,
-                $methodAst,
-                $method->getDeclaringClass()->getDeclaringNamespaceAst(),
-                $method->getDeclaringClass(),
-                $this,
-                $aliasMethodName,
-            );
-        }
-
-        if ($aliases !== []) {
-            return $aliases;
-        }
-
-        if (array_key_exists($methodHash, $traitPrecedences)) {
-            return [];
-        }
-
-        $newMethod = ReflectionMethod::createFromNode(
-            $this->reflector,
-            $methodAst,
-            $method->getDeclaringClass()->getDeclaringNamespaceAst(),
-            $method->getDeclaringClass(),
-            $this,
-            $method->getAliasName(),
+        return array_merge(
+            [],
```

Comment:
BTW, since we are on PHP 7.4+, `array_merge([], ...$others)` is no longer needed: https://3v4l.org/HUtnH

Reply:
Not relevant for this patch: maybe interesting for slevomat/coding-standard :D